In [ ]:
# utilizado para manejos de directorios y rutas
import os

# Computacion vectorial y cientifica para python
import numpy as np

# Librerias para graficación (trazado de gráficos)
from matplotlib import pyplot
from mpl_toolkits.mplot3d import Axes3D  # Necesario para graficar superficies 3D
# Modulo de optimización
from scipy import optimize

# Modulo para cargar archivos de datos en formato MATLAB
from scipy.io import loadmat


# llama a matplotlib a embeber graficas dentro de los cuadernillos
%matplotlib inline
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from datetime import datetime


In [ ]:
from google.colab import drive
drive.mount("/content/gdrive")

Mounted at /content/gdrive


In [ ]:
!ls
%mkdir data
!ls

data  gdrive  sample_data
mkdir: cannot create directory ‘data’: File exists
data  gdrive  sample_data


In [91]:
#  datos de entrenamiento almacenados en los arreglos X, y
# data = np.loadtxt("/content/gdrive/MyDrive/Colab Notebooks/machine learning/datasets/wine_preparado.csv", delimiter=',')
data = pd.read_csv('/content/gdrive/MyDrive/IA/DATASET/16P.csv', header=None, encoding='latin1',skiprows=1)



# print(data)

# Dividir los datos en características (X) y etiquetas (y)
X = data.iloc[:, 1:-1]  # Seleccionar todas las filas y todas las columnas desde la segunda hasta la penúltima
y = data.iloc[:, -1]    # Seleccionar todas las filas y la última columna


#X, y = data[:, 1:61], data[:,61]
# Definir el diccionario de mapeo para convertir letras a números
mapeo_y = {'INTJ': 1, 'INTP': 2, 'ENTJ': 3, 'ENTP': 4,
           'INFJ': 5, 'INFP': 6, 'ENFJ': 7, 'ENFP': 8,
           'ISTJ': 9, 'ISFJ': 10, 'ESTJ': 11, 'ESFJ': 12,
           'ISTP': 13, 'ISFP': 14, 'ESTP': 15, 'ESFP': 16}

# Convertir las letras a números utilizando replace
data.replace(mapeo_y, inplace=True)
y = np.array([int(e) for e in y])
print(y.shape)
y = np.squeeze(y)

y[y == 1] = 0
y[y == 2] = 1
y[y == 3] = 2
y[y == 4] = 3
y[y == 5] = 4
y[y == 6] = 5
y[y == 7] = 6
y[y == 8] = 7
y[y == 9] = 8
y[y == 10] = 9
y[y == 11] = 10
y[y == 12] = 11
y[y == 13] = 12
y[y == 14] = 13
y[y == 15] = 14
y[y == 16] = 15

Xp = X.iloc[50000:, :]

#Xp= X[50000:,:]
X = X.iloc[:50000,:]


# Imprimir algunas filas de datos
print('{:>8s}{:>8s}{:>10s}{:>10s}{:>8s}{:>8s}{:>8s}{:>8s}{:>6s}{:>10s}{:>10s}{:>10s}'.format('X[:,0]', 'X[:, 1]', 'X[:, 2]', 'X[:, 3]', 'X[:, 4]', 'X[:, 5]', 'X[:, 6]', 'X[:, 7]', 'X[:, 8]', 'X[:, 9]', 'X[:, 10]', 'Y'
))
print('-'*50)
for i in range(500):
    print('{:8.0f}{:8.0f}{:10.0f}{:10.0f}{:8.0f}{:8.0f}{:8.0f}{:8.0f}{:8.0f}{:10.0f}{:8.0f}{:12.0f}'.format( X.iloc[i, 0], X.iloc[i, 1], X.iloc[i, 2], X.iloc[i, 3], X.iloc[i, 4], X.iloc[i, 5], X.iloc[i, 6], X.iloc[i, 7], X.iloc[i, 8], X.iloc[i, 9], X.iloc[i, 10], y[i]))
yp = y[50000:]
y = y[:50000]
print("la cantidad de elementos de x entrenamiento: ")
print(X.shape)
print("la cantidad de elementos de y entrenamiento: ")
print(y.shape)
print("la cantidad de elementos de x prueba: ")
print(Xp.shape)
print("la cantidad de elementos dey prueba: ")
print(yp.shape)
print(yp)


(59999,)
  X[:,0] X[:, 1]   X[:, 2]   X[:, 3] X[:, 4] X[:, 5] X[:, 6] X[:, 7]X[:, 8]   X[:, 9]  X[:, 10]         Y
--------------------------------------------------
       0       0         0         0       0       1       1       0       0         0       0           7
       0       0        -2        -3      -1       2      -2       0       3         0      -2          13
       0       0         2         0      -1       2       0       0       1         0       0           4
       0      -1         3        -1       0       0      -2       0      -2         0       2          12
       0       0        -1         0       2      -1      -2       0       1         0       0           6
       0       0         1         0       1      -2      -1       0       2         1       1           0
       0       0        -2         0      -1       1      -2       0       1         1      -1          13
       0       0        -3         0      -1       0      -2       0       0         

## **Definimos las capas**
En este apartado definiremos la cantidad de capas que tendra el modelo:

Capa de entrada: Es la cantidad de caracteristicas con las que cuenta el dataset.

Capa oculta: esto se define identificando cual seria el mejor valor

Capa de salida:En mi caso son 16 valores de salida

In [ ]:
# Configurando parametros necesario
input_layer_size  =  60 # Entrada de 60 caracteristicas
hidden_layer_size = 40   # 40 unidades ocultas
num_labels = 16        # 16etiquetas,

# carga los pesos en las variables Theta1 y Theta2
# weights = loadmat(os.path.join('/content/gdrive/MyDrive/Colab Notebooks/machine learning/data', 'ex4weights.mat'))
# weights = np.array()
pesos = {}
pesos['Theta1'] = np.random.rand(40, 61)
pesos['Theta2'] = np.random.rand(16, 41)

Theta1, Theta2 = pesos['Theta1'], pesos['Theta2']


# Desenrollar parámetros
print(Theta1.ravel().shape)
print(Theta2.ravel().shape)

nn_params = np.concatenate([Theta1.ravel(), Theta2.ravel()])#Concadenamos Theta1 y Theta2 para formar un solo array
print(nn_params.shape)

(2440,)
(656,)
(3096,)


In [ ]:
def sigmoid(z):
    """
    Computes the sigmoid of z.
    """
    return 1.0 / (1.0 + np.exp(-z))


def sigmoidGradient(z):

    g = np.zeros(z.shape)

    g = sigmoid(z) * (1 - sigmoid(z))

    return g

### **CALCULAR COSTO**

In [ ]:
def nnCostFunction(nn_params,
                   input_layer_size,
                   hidden_layer_size,
                   num_labels,
                   X, y, lambda_= 0.0):

    # Reshape nn_params back into the parameters Theta1 and Theta2, the weight matrices
    # for our 2 layer neural network
    Theta1 = np.reshape(nn_params[:hidden_layer_size * (input_layer_size + 1)],
                        (hidden_layer_size, (input_layer_size + 1)))

    Theta2 = np.reshape(nn_params[(hidden_layer_size * (input_layer_size + 1)):],
                        (num_labels, (hidden_layer_size + 1)))

    m = y.size

    J = 0
    Theta1_grad = np.zeros(Theta1.shape)
    Theta2_grad = np.zeros(Theta2.shape)

    a1 = np.concatenate([np.ones((m, 1)), X], axis=1)

    a2 = sigmoid(a1.dot(Theta1.T))
    a2 = np.concatenate([np.ones((a2.shape[0], 1)), a2], axis=1)

    a3 = sigmoid(a2.dot(Theta2.T))


    y_matrix = y.reshape(-1)
    # print(y.shape)
    y_matrix = np.eye(num_labels)[y_matrix]
    # print(y_matrix)

    temp1 = Theta1
    temp2 = Theta2

    # Agregar el termino de regularización

    reg_term = (lambda_ / (2 * m)) * (np.sum(np.square(temp1[:, 1:])) + np.sum(np.square(temp2[:, 1:])))

    J = (-1 / m) * np.sum((np.log(a3) * y_matrix) + np.log(1 - a3) * (1 - y_matrix)) + reg_term

    # Backpropogation

    delta_3 = a3 - y_matrix
    delta_2 = delta_3.dot(Theta2)[:, 1:] * sigmoidGradient(a1.dot(Theta1.T))

    Delta1 = delta_2.T.dot(a1)
    Delta2 = delta_3.T.dot(a2)

    # Agregar regularización al gradiente

    Theta1_grad = (1 / m) * Delta1
    Theta1_grad[:, 1:] = Theta1_grad[:, 1:] + (lambda_ / m) * Theta1[:, 1:]

    Theta2_grad = (1 / m) * Delta2
    Theta2_grad[:, 1:] = Theta2_grad[:, 1:] + (lambda_ / m) * Theta2[:, 1:]

    # ===================== Alterntate solutions =====================
    # my_final_matrix = np.zeros(a3.shape)
    # for c in np.arange(num_labels):
    #    my_final_matrix[:, c] = (np.log(a3[:, c]) * (y == c)) + (np.log(1 - a3[:, c]) * (1 - (y == c)))
    #J = (-1 / m) * np.sum(my_final_matrix)
    # ================================================================

    # ================================================================
    # Unroll gradients
    # grad = np.concatenate([Theta1_grad.ravel(order=order), Theta2_grad.ravel(order=order)])

    grad = np.concatenate([Theta1_grad.ravel(), Theta2_grad.ravel()])

    return J, grad

In [ ]:
lambda_ = 0
J, _ = nnCostFunction(nn_params, input_layer_size, hidden_layer_size, num_labels, X, y, lambda_)
print('Costo en parametros (cargado de ex4weights): %.6f ' % J)
print('El costo debe esta cercano a               : 0.287629')

Costo en parametros (cargado de ex4weights): 154.638047 
El costo debe esta cercano a               : 0.287629


In [ ]:
z = np.array([-1, -0.5, 0, 0.5, 1])
g = sigmoidGradient(z)
print('Gradiente sigmoide evaluada con [-1 -0.5 0 0.5 1]:\n  ')
print(g)

Gradiente sigmoide evaluada con [-1 -0.5 0 0.5 1]:
  
[0.19661193 0.23500371 0.25       0.23500371 0.19661193]


In [ ]:
def randInitializeWeights(L_in, L_out, epsilon_init=0.12):
    """
    Randomly initialize the weights of a layer in a neural network.

    Parameters
    ----------
    L_in : int
        Number of incomming connections.

    L_out : int
        Number of outgoing connections.

    epsilon_init : float, optional
        Range of values which the weight can take from a uniform
        distribution.

    Returns
    -------
    W : array_like
        The weight initialiatized to random values.  Note that W should
        be set to a matrix of size(L_out, 1 + L_in) as
        the first column of W handles the "bias" terms."""


    W = np.zeros((L_out, 1 + L_in))
    W = np.random.rand(L_out, 1 + L_in) * 2 * epsilon_init - epsilon_init

    return W

In [ ]:
print('Inicialización de parámetros de redes neuronales...')

initial_Theta1 = randInitializeWeights(input_layer_size, hidden_layer_size)
initial_Theta2 = randInitializeWeights(hidden_layer_size, num_labels)

# Desenrrollr parametros
initial_nn_params = np.concatenate([initial_Theta1.ravel(), initial_Theta2.ravel()], axis=0)

Inicialización de parámetros de redes neuronales...


In [ ]:
#  After you have completed the assignment, change the maxiter to a larger
#  value to see how more training helps.
options= {'maxiter': 1000}

#  You should also try different values of lambda
lambda_ = 1

# Create "short hand" for the cost function to be minimized
costFunction = lambda p: nnCostFunction(p, input_layer_size,
                                        hidden_layer_size,
                                        num_labels, X, y, lambda_)

# Now, costFunction is a function that takes in only one argument
# (the neural network parameters)
res = optimize.minimize(costFunction,
                        initial_nn_params,
                        jac=True,
                        method='TNC',
                        options=options)

# get the solution of the optimization
nn_params = res.x

# Obtain Theta1 and Theta2 back from nn_params
Theta1 = np.reshape(nn_params[:hidden_layer_size * (input_layer_size + 1)],
                    (hidden_layer_size, (input_layer_size + 1)))

Theta2 = np.reshape(nn_params[(hidden_layer_size * (input_layer_size + 1)):],
                    (num_labels, (hidden_layer_size + 1)))

<ipython-input-157-2f5ec00f0be7>:15: OptimizeWarning: Unknown solver options: maxiter
  res = optimize.minimize(costFunction,
<ipython-input-152-de0ba5c46d6a>:44: RuntimeWarning: divide by zero encountered in log
  J = (-1 / m) * np.sum((np.log(a3) * y_matrix) + np.log(1 - a3) * (1 - y_matrix)) + reg_term
<ipython-input-152-de0ba5c46d6a>:44: RuntimeWarning: invalid value encountered in multiply
  J = (-1 / m) * np.sum((np.log(a3) * y_matrix) + np.log(1 - a3) * (1 - y_matrix)) + reg_term


In [ ]:
def predict(Theta1, Theta2, X):
    """
    Predict the label of an input given a trained neural network
    Outputs the predicted label of X given the trained weights of a neural
    network(Theta1, Theta2)
    """
    # Useful values
    m = X.shape[0]
    num_labels = Theta2.shape[0]

    # You need to return the following variables correctly
    p = np.zeros(m)
    h1 = sigmoid(np.dot(np.concatenate([np.ones((m, 1)), X], axis=1), Theta1.T))
    h2 = sigmoid(np.dot(np.concatenate([np.ones((m, 1)), h1], axis=1), Theta2.T))
    p = np.argmax(h2, axis=1)
    return p

In [ ]:
pred = predict(Theta1, Theta2, X.iloc[:,:])
print(pred)
print('Training Set Accuracy: %f' % (np.mean(pred == y[:]) * 100))

[ 7 13  4 ...  0 13  5]
Training Set Accuracy: 96.824000


In [ ]:
pred = predict(Theta1, Theta2, Xp.iloc[:,:])
print(pred)
print('Training Set Accuracy: %f' % (np.mean(pred == yp[:]) * 100))


[ 3 14  1 ... 12  8  4]
Training Set Accuracy: 95.759576


Con este modelo se obtuvo buenos resultados de un 96%  de entrenamiento y un 95% de prueba. Esto indica que el modelo está generalizando bien y no está sufriendo de sobreajuste significativo.

### **REDES NEURONALES CON PYTORCH**

---



In [ ]:
# utilizado para la manipulación de directorios y rutas
import os
# Cálculo científico y vectorial para python
import numpy as np
# Libreria para graficos
from matplotlib import pyplot as plt
import pandas as pd

import torch
from torch import optim  # For optimizers like SGD, Adam, etc.
from torch import nn  # All neural network modules
from torch.utils.data import DataLoader  # Gives easier dataset managment by creating mini batches etc.

import torchvision # torch package for vision related things
import torchvision.transforms as transforms  # Transformations we can perform on our dataset for augmentation
import torch.nn.functional as F  # Parameterless functions, like (some) activation functions
import torchvision.datasets as datasets  # Standard datasets
from tqdm import tqdm  # For nice progress bar!
from torch.utils.data import random_split, TensorDataset

# le dice a matplotlib que incruste gráficos en el cuaderno
%matplotlib inline

In [ ]:
#  datos de entrenamiento almacenados en los arreglos X, y
datos = pd.read_csv('/content/gdrive/MyDrive/IA/DATASET/16P.csv', header=None, encoding='latin1',skiprows=1)


In [188]:

# Separamos la ultima columna para que sea variable destino
datos_y = datos[datos.columns[-1]]

# Definir el diccionario de mapeo para convertir letras a números
mapeo_y = {'INTJ': 1, 'INTP': 2, 'ENTJ': 3, 'ENTP': 4,
           'INFJ': 5, 'INFP': 6, 'ENFJ': 7, 'ENFP': 8,
           'ISTJ': 9, 'ISFJ': 10, 'ESTJ': 11, 'ESFJ': 12,
           'ISTP': 13, 'ISFP': 14, 'ESTP': 15, 'ESFP': 16}

# Convertir las letras a números utilizando replace
datos.replace(mapeo_y, inplace=True)
#datos_y= np.array([int(e) for e in y])
datos_y = np.array([int(e) for e in datos_y])
print(datos_y.shape)
datos_y = np.squeeze(datos_y)
# Ajustar las etiquetas para que estén en el rango de 0 a 15
datos_y[datos_y == 1] = 0
datos_y[datos_y == 2] = 1
datos_y[datos_y == 3] = 2
datos_y[datos_y == 4] = 3
datos_y[datos_y == 5] = 4
datos_y[datos_y == 6] = 5
datos_y[datos_y == 7] = 6
datos_y[datos_y == 8] = 7
datos_y[datos_y == 9] = 8
datos_y[datos_y == 10] = 9
datos_y[datos_y == 11] = 10
datos_y[datos_y == 12] = 11
datos_y[datos_y == 13] = 12
datos_y[datos_y == 14] = 13
datos_y[datos_y == 15] = 14
datos_y[datos_y == 16] = 15





(59999,)


In [189]:
print(datos_y)

[ 7 13  4 ... 12  8  4]


In [190]:
datos_x= datos.iloc[:, 1:-1]
#datos_x = datos.drop(columns=[0,61])#eliminamos la colum 0 por q son ids y la columna de la salida
datos_x.head()

1   2   3   4   5   6   7   8   9   10  ...  51  52  53  54  55  56  57  \
0   0   0   0   0   0   1   1   0   0   0  ...   0   0   0   0  -1   0   0   
1   0   0  -2  -3  -1   2  -2   0   3   0  ...   0   0  -2   0   2   0  -1   
2   0   0   2   0  -1   2   0   0   1   0  ...   0   0   2   0   2  -1   0   
3   0  -1   3  -1   0   0  -2   0  -2   0  ...   0   0   0  -1  -1   0   1   
4   0   0  -1   0   2  -1  -2   0   1   0  ...   0   0   1   0   2   0   1   

   58  59  60  
0   0   0   0  
1  -1  -1   3  
2   1   2   1  
3   0  -2  -1  
4  -1   2  -1  

[5 rows x 60 columns]

PARA DIVIDIR EN ENTRENAMIENTO Y PRUEBA

In [191]:
datos_x.shape[0]

59999

In [192]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(datos_x, datos_y, test_size=0.2, random_state=2)



In [193]:
print("X Train: {}, X Test: {}, y_train: {}, y_test: {}".format(X_train.shape, X_test.shape, y_train.shape, y_test.shape))

X Train: (47999, 60), X Test: (12000, 60), y_train: (47999,), y_test: (12000,)


In [194]:

# Set device cuda for GPU if it's available otherwise run on the CPU
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print(device)

# Hyperparameters of our neural network which depends on the dataset, and
# also just experimenting to see what works well (learning rate for example).
input_size =  X_train.shape[1]
num_classes = 16
learning_rate = 0.001
batch_size = 10000
num_epochs = 4

cpu


In [195]:
from torch.utils.data import TensorDataset, DataLoader
import torch

# Convertir a tensores de PyTorch
t_X_train = torch.tensor(X_train.values, dtype=torch.float32)
t_X_test = torch.tensor(X_test.values, dtype=torch.float32)
t_y_train = torch.tensor(y_train, dtype=torch.long)
t_y_test = torch.tensor(y_test, dtype=torch.long)

# Crear conjuntos de datos con TensorDataset
train_dataset = TensorDataset(t_X_train, t_y_train)
test_dataset = TensorDataset(t_X_test, t_y_test)

# Definir el tamaño del lote y crear DataLoader para entrenamiento y prueba
batch_size = 64
train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
test_loader = DataLoader(test_dataset, batch_size=batch_size, shuffle=False)


In [196]:
test = TensorDataset(t_X_test, t_y_test)
print(test[0])

(tensor([ 0.,  0.,  3.,  2., -3., -2., -1.,  0.,  2.,  0.,  0., -2.,  0.,  3.,
         1., -1.,  2.,  1., -1.,  0.,  0.,  1.,  1.,  0., -2.,  1., -1.,  0.,
         1.,  3., -1.,  0.,  3.,  0., -2.,  0., -1.,  0., -1.,  2., -1., -1.,
        -1.,  0., -2.,  0., -1.,  0.,  1.,  0.,  0.,  0.,  3.,  0.,  1.,  0.,
         0.,  1.,  3.,  2.]), tensor(0))


In [197]:
t_y_train

tensor([ 9,  2,  2,  ..., 12, 10,  5])

In [198]:
# Here we create our simple neural network. For more details here we are subclassing and
# inheriting from nn.Module, this is the most general way to create your networks and
# allows for more flexibility. I encourage you to also check out nn.Sequential which
# would be easier to use in this scenario but I wanted to show you something that
# "always" works.
class RedNeuronalMLS(nn.Module):
    def __init__(self, input_size, num_classes):
        super(RedNeuronalMLS, self).__init__()
        # Our first linear layer take input_size, in this case 784 nodes to 50
        # and our second linear layer takes 50 to the num_classes we have, in
        # this case 10.
        self.fc1 = nn.Linear(input_size, 40)
        self.fc2 = nn.Linear(40, num_classes)

    def forward(self, x):
        """
        x here is the mnist images and we run it through fc1, fc2 that we created above.
        we also add a ReLU activation function in between and for that (since it has no parameters)
        I recommend using nn.functional (F)
        """
        x = self.fc1(x)
        x = F.sigmoid(x)
        # x = F.sigmoid(self.fc1(x))
        x = self.fc2(x)
        return x

In [199]:
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
# Initialize network
model = RedNeuronalMLS(input_size=input_size, num_classes=num_classes).to(device)

In [200]:
model

RedNeuronalMLS(
  (fc1): Linear(in_features=60, out_features=40, bias=True)
  (fc2): Linear(in_features=40, out_features=16, bias=True)
)

In [201]:
# Loss and optimizer
criterion = nn.CrossEntropyLoss()
optimizer = optim.Adam(model.parameters(), lr=learning_rate)

In [203]:
# Train Network
for epoch in range(num_epochs):
    for batch_idx, (data, targets) in enumerate(tqdm(train_loader)):
    # for batch_idx, (data, targets) in enumerate(train_loader):

        # Get data to cuda if possible
        data = data.to(device=device)
        targets = targets.to(device=device)

        # print(data.shape)
        # Get to correct shape
        data = data.reshape(data.shape[0], -1)
        print(data.shape)
        # print("-"*30)
        # forward
        scores = model(data)
        loss = criterion(scores, targets)

        # backward
        optimizer.zero_grad()
        loss.backward()

        # gradient descent or adam step
        optimizer.step()

  9%|▉         | 69/750 [00:00<00:02, 340.27it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 14%|█▍        | 104/750 [00:00<00:01, 326.53it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 23%|██▎       | 170/750 [00:00<00:01, 317.84it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 33%|███▎      | 244/750 [00:00<00:01, 345.83it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 42%|████▏     | 314/750 [00:00<00:01, 341.58it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 51%|█████     | 383/750 [00:01<00:01, 332.67it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 61%|██████    | 455/750 [00:01<00:00, 343.12it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 70%|███████   | 526/750 [00:01<00:00, 328.17it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 79%|███████▉  | 595/750 [00:01<00:00, 334.53it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 89%|████████▉ | 667/750 [00:01<00:00, 344.81it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

100%|██████████| 750/750 [00:02<00:00, 339.60it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

  0%|          | 0/750 [00:00<?, ?it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 10%|▉         | 72/750 [00:00<00:01, 355.78it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 20%|█▉        | 147/750 [00:00<00:01, 364.27it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 30%|██▉       | 222/750 [00:00<00:01, 362.64it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 41%|████      | 304/750 [00:00<00:01, 387.73it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 46%|████▌     | 343/750 [00:00<00:01, 373.55it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 51%|█████     | 381/750 [00:01<00:01, 365.15it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 56%|█████▌    | 419/750 [00:01<00:00, 368.49it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 61%|██████    | 459/750 [00:01<00:00, 377.42it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 66%|██████▋   | 497/750 [00:01<00:00, 360.90it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 72%|███████▏  | 537/750 [00:01<00:00, 370.44it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 77%|███████▋  | 575/750 [00:01<00:00, 368.43it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 82%|████████▏ | 612/750 [00:01<00:00, 365.48it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 87%|████████▋ | 650/750 [00:01<00:00, 368.38it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 92%|█████████▏| 688/750 [00:01<00:00, 370.67it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


100%|██████████| 750/750 [00:02<00:00, 364.46it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

  0%|          | 0/750 [00:00<?, ?it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 10%|▉         | 72/750 [00:00<00:01, 356.79it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 14%|█▍        | 108/750 [00:00<00:02, 310.08it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 19%|█▉        | 145/750 [00:00<00:01, 328.70it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 24%|██▍       | 181/750 [00:00<00:01, 338.61it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 29%|██▉       | 219/750 [00:00<00:01, 349.20it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 34%|███▍      | 255/750 [00:00<00:01, 347.55it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 39%|███▊      | 290/750 [00:00<00:01, 341.07it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 43%|████▎     | 325/750 [00:00<00:01, 338.63it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 48%|████▊     | 362/750 [00:01<00:01, 345.06it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 53%|█████▎    | 397/750 [00:01<00:01, 342.36it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 58%|█████▊    | 433/750 [00:01<00:00, 345.98it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 62%|██████▏   | 468/750 [00:01<00:00, 341.65it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 67%|██████▋   | 506/750 [00:01<00:00, 351.70it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 72%|███████▏  | 542/750 [00:01<00:00, 349.84it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 77%|███████▋  | 578/750 [00:01<00:00, 351.85it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 82%|████████▏ | 615/750 [00:01<00:00, 355.94it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 87%|████████▋ | 651/750 [00:01<00:00, 354.90it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 92%|█████████▏| 687/750 [00:01<00:00, 350.95it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([6

 96%|█████████▋| 723/750 [00:02<00:00, 350.22it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


100%|██████████| 750/750 [00:02<00:00, 346.09it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([63, 60])


  0%|          | 0/750 [00:00<?, ?it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


  5%|▌         | 39/750 [00:00<00:01, 378.28it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 10%|█         | 77/750 [00:00<00:02, 325.39it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 15%|█▍        | 110/750 [00:00<00:02, 306.74it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 19%|█▉        | 141/750 [00:00<00:01, 305.29it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 24%|██▎       | 177/750 [00:00<00:01, 321.62it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 29%|██▊       | 214/750 [00:00<00:01, 336.53it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 33%|███▎      | 248/750 [00:00<00:01, 262.72it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 37%|███▋      | 277/750 [00:00<00:01, 249.83it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 41%|████      | 304/750 [00:01<00:01, 239.98it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 44%|████▍     | 330/750 [00:01<00:01, 243.66it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 47%|████▋     | 356/750 [00:01<00:01, 219.15it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 51%|█████▏    | 385/750 [00:01<00:01, 235.09it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 55%|█████▌    | 414/750 [00:01<00:01, 245.71it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 59%|█████▊    | 440/750 [00:01<00:01, 241.75it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 62%|██████▏   | 465/750 [00:01<00:01, 240.23it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 65%|██████▌   | 490/750 [00:01<00:01, 242.51it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 69%|██████▉   | 517/750 [00:01<00:00, 248.14it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 73%|███████▎  | 544/750 [00:02<00:00, 253.56it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 76%|███████▌  | 570/750 [00:02<00:00, 254.99it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 79%|███████▉  | 596/750 [00:02<00:00, 232.37it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 83%|████████▎ | 620/750 [00:02<00:00, 228.33it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 86%|████████▌ | 644/750 [00:02<00:00, 228.58it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


 89%|████████▉ | 669/750 [00:02<00:00, 233.14it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 92%|█████████▏| 693/750 [00:02<00:00, 234.77it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])

 96%|█████████▌| 720/750 [00:02<00:00, 244.56it/s]


torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])


100%|██████████| 750/750 [00:02<00:00, 253.16it/s]

torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([64, 60])
torch.Size([63, 60])


In [204]:
# Check accuracy on training & test to see how good our model
def check_accuracy(loader, model):
    num_correct = 0
    num_samples = 0
    model.eval()

    predicciones = []
    with torch.no_grad():
        for x, y in loader:
            x = x.to(device=device)
            y = y.to(device=device)
            x = x.reshape(x.shape[0], -1)

            scores = model(x)
            _, predictions = scores.max(1)
            predicciones.append(predictions)

            num_correct += (predictions == y).sum()
            num_samples += predictions.size(0)

    model.train()
    return num_correct/num_samples, predicciones

p_train, pred_train  = check_accuracy(train_loader, model)
p_test, pred_test  = check_accuracy(test_loader, model)

print(f"Accuracy on training set: {p_train*100:.2f}")
print(f"Accuracy on test set: {p_test*100:.2f}")

Accuracy on training set: 94.58
Accuracy on test set: 93.55


Con PyTorch, tienes un mayor control sobre el proceso de construcción y entrenamiento de la red neuronal pero solo se obtuvo un 94% de entrenamiento y un 95% de testing lo cual quiere decir q obtuve mejores resultados con el anterior modelo